src/PreValidatableRuleInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator;
+
+use Closure;
+
+interface PreValidatableRuleInterface
+{
+    public function isSkipOnEmpty(): bool;
+
+    public function isSkipOnError(): bool;
```

Comment:
This way it may be better English:

```suggestion
    public function shouldSkipOnEmpty(): bool;

    public function shouldSkipOnError(): bool;
```